python/ray/ray_constants.py

```diff
@@ -49,7 +53,47 @@ def env_integer(key, default):
 # The maximum resource quantity that is allowed. TODO(rkn): This could be
 # relaxed, but the current implementation of the node manager will be slower
 # for large resource quantities due to bookkeeping of specific resource IDs.
-MAX_RESOURCE_QUANTITY = 512
+MAX_RESOURCE_QUANTITY = 10000
+
+# Each memory "resource" counts as this many bytes of memory.
+MEMORY_RESOURCE_UNIT_BYTES = 100 * 1024 * 1024
+
+# Number of units 1 resource can be subdivided into.
+MIN_RESOURCE_GRANULARITY = 0.0001
+
+# Fraction of plasma memory that can be reserved. It is actually 70% but this
+# is set to 69% to leave some headroom.
+PLASMA_RESERVABLE_MEMORY_FRACTION = 0.69
```

Comment:
```python
import ray

ray.init(object_store_memory=10**9)

@ray.remote(object_store_memory=3.4*10**8)
class Foo:
    def __init__(self):
        print("XXX")

f1 = Foo.remote()  # Prints "XXX"
f2 = Foo.remote()  # Doesn't print anything.
```

Shouldn't I be able to create two copies of the above class? Somehow only one can be created at a time.

Reply:
You can only schedule up to 70% of the object store memory unfortunately, which explains this behavior.